tests/integration/automated_chatflows/test_solutions.py

```diff
@@ -0,0 +1,311 @@
+        cls.ssh_cl.private_key_path = "/tmp/.ssh/id_rsa"
+        cls.ssh_cl.load_from_file_system()
+        cls.ssh_cl.save()
+        cls.solution_uuid = ""
+
+    @classmethod
+    def tearDownClass(cls):
+        # should stop threebot server.
+        j.sals.process.execute(f"sudo wg-quick down /tmp/{cls.wg_name}.conf")
+        j.sals.fs.rmtree(path=f"/tmp/{cls.wg_name}.conf")
+        j.sals.fs.rmtree(path=f"/tmp/.ssh")
+        j.clients.sshkey.delete(cls.ssh_client_name)
+
+    def tearDown(self):
+        if self.solution_uuid:
+            j.sals.reservation_chatflow.solutions.cancel_solution_by_uuid(self.solution_uuid)
+
+        j.clients.sshclient.delete(self.ssh_client_name)
+
+    def test01_ubuntu(self):
+        """Test case for create ubuntu.
+
+        **Test Scenario**
+        #. create ubuntu
+        #. connect to ubuntu
+        #. check that ubuntu is accessed
+        """
+
+        base.info("create ubuntu.")
+        name = base.random_string()
+        ubuntu = deployer.deploy_ubuntu(solution_name=name, network=self.network_name, ssh=self.ssh_cl.public_key_path)
+        self.solution_uuid = ubuntu.solution_id
+
+        base.info("connect to ubuntu")
+        localclient = j.clients.sshclient.get(self.ssh_client_name)
+        localclient.sshkey = self.ssh_client_name
+        localclient.host = ubuntu.ip_address
+        localclient.save()
+        self.solution_uuid = ubuntu.solution_id
+
+        base.info("check that ubuntu is accessed")
+        import pdb
+
+        pdb.set_trace()
+        self.assertTrue(
+            j.sals.nettools.tcp_connection_test(ubuntu.ip_address, port=22, timeout=40),
+            "Ubuntu is not reached after 30 second",
+        )
+        _, res, _ = localclient.sshclient.run("cat /etc/os-release")
+        self.assertIn("successfully", ubuntu.success())
+        self.assertIn('VERSION_ID="18.04"', res)
+
+    def test02_kubernetes(self):
+        """Test case for create kubernetes.
+
+        **Test Scenario**
+        #. create kubernetes
+        #. check connection to master
+        """
+        base.info("create kubernetes")
+        name = base.random_string()
+        secret = base.random_string()
+        kubernetes = deployer.deploy_kubernetes(
+            solution_name=name, secret=secret, network=self.network_name, ssh=self.ssh_cl.public_key_path,
+        )
+        self.solution_uuid = kubernetes.solution_id
+
+        localclient = j.clients.sshclient.get(self.ssh_client_name)
+        localclient.sshkey = self.ssh_client_name
+        localclient.host = f"{kubernetes.ip_addresses[0]}"
+        localclient.user = "rancher"
+        localclient.save()
+        self.assertTrue(
+            j.sals.nettools.tcp_connection_test(kubernetes.ip_addresses[0], port=22, timeout=40),
+            "master is not reached after 30 second",
+        )
+        _, res, _ = localclient.sshclient.run("kubectl get nodes")
+        self.assertIn(3, len(res.splitlines()))
```

Comment:
can you pass the number of workers deployed to deploy method, then here check them here, as `3` don't mean anything for who is reading the test.